test/test_regression.cpp

```diff
@@ -0,0 +1,310 @@
+///////////////////////////////////////////////////////////////////////////////
+// File:         test_regression.cpp
+// Description:  Auxiliar independent script to perform regression
+// Authors:      Florensa, Miquel & Luong-Ha Nguyen & James-A. Goulet
+// Created:      February 20, 2023
+// Updated:      February 20, 2023
+// Contact:      miquelflorensa11@gmail.com & luongha.nguyen@gmail.com &
+//               james.goulet@polymtl.ca
+// Copyright (c) 2022 Luong-Ha Nguyen & James-A. Goulet.
+// Some rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#include "test_regression.h"
+
+/**
+ * @brief Perform a linear regression on the train data.
+ *
+ * @param[out] net the network to train with specified architecture and
+ * parameters
+ * @param[in] db the database to train the network on
+ */
+void regression_train(TagiNetworkCPU &net, Dataloader &db, int epochs) {
+    // Number of data points
+    int n_iter = db.num_data / net.prop.batch_size;
+    std::vector<int> data_idx = create_range(db.num_data);
+
+    std::default_random_engine seed_e(0);
+
+    // Initialize the data's variables
+    std::vector<float> x_batch(net.prop.batch_size * net.prop.n_x, 0);
+    std::vector<float> Sx_batch(net.prop.batch_size * net.prop.n_x,
+                                pow(net.prop.sigma_x, 2));
+    std::vector<float> Sx_f_batch;
+    std::vector<float> y_batch(net.prop.batch_size * net.prop.n_y, 0);
+    std::vector<float> V_batch(net.prop.batch_size * net.prop.n_y,
+                               pow(net.prop.sigma_v, 2));
+    std::vector<int> batch_idx(net.prop.batch_size);
+    std::vector<int> idx_ud_batch(net.prop.nye * net.prop.batch_size, 0);
+
+    // *TODO: Is there any better way?
+    if (net.prop.is_full_cov) {
+        float var_x = pow(net.prop.sigma_x, 2);
+        auto Sx_f = initialize_upper_triu(var_x, net.prop.n_x);
+        Sx_f_batch = repmat_vector(Sx_f, net.prop.batch_size);
+    }
+
+    for (int e = 0; e < epochs; e++) {
+        if (e > 0) {
```

Comment:
Please remove line 48 to 55. We don't want to shuffle data randomly when generating the outputs for unittest because the test will be performed in different OS. It should be static for the comparison purpose.